wpilibj/src/main/java/edu/wpi/first/wpilibj/ResistanceCalculator.java

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj;
+
+import edu.wpi.first.util.CircularBuffer;
+
+/**
+ * Finds the resistance of a channel using a running linear regression over a window. Must be
+ * updated with current and voltage periodically using the {@link
+ * ResistanceCalculator#calculate(double, double) calculate} method.
+ *
+ * <p>To use this for finding the resistance of a channel, use the calculate method with the battery
+ * voltage minus the voltage at the motor controller or whatever is plugged in to the PDP at that
+ * channel.
+ */
+public class ResistanceCalculator {
+  public static final int kDefaultBufferSize = 250;
+  public static final double kDefaultRSquaredThreshold = 0.75;
+
+  /**
+   * Buffers holding the current values that will eventually need to be subtracted from the sum when
+   * they leave the window.
+   */
+  private final CircularBuffer m_currentBuffer;
+  /**
+   * Buffer holding the voltage values that will eventually need to be subtracted from the sum when
+   * they leave the window.
+   */
+  private final CircularBuffer m_voltageBuffer;
+  /** The maximum number of points to take the linear regression over. */
+  private final int m_bufferSize;
+  /**
+   * The minimum R^2 value considered significant enough to return the regression slope instead of
+   * NaN.
+   */
+  private final double m_rSquaredThreshold;
+  /** Running sum of the past currents. */
+  private double m_currentSum;
+  /** Running sum of the past voltages. */
+  private double m_voltageSum;
+  /** Running sum of the squares of the past currents. */
+  private double m_currentSquaredSum;
+  /** Running sum of the squares of the past voltages. */
+  private double m_voltageSquaredSum;
+  /** Running sum of the past current*voltage's. */
+  private double m_prodSum;
+  /** The number of points currently in the buffer. */
```

Comment:
```suggestion
  private final CircularBuffer m_voltageBuffer;

  /** The maximum number of points to take the linear regression over. */
  private final int m_bufferSize;

  /**
   * The minimum R^2 value considered significant enough to return the regression slope instead of
   * NaN.
   */
  private final double m_rSquaredThreshold;

  /** Running sum of the past currents. */
  private double m_currentSum;

  /** Running sum of the past voltages. */
  private double m_voltageSum;

  /** Running sum of the squares of the past currents. */
  private double m_currentSquaredSum;

  /** Running sum of the squares of the past voltages. */
  private double m_voltageSquaredSum;

  /** Running sum of the past current*voltage's. */
  private double m_prodSum;

  /** The number of points currently in the buffer. */
```